client/src/Views/Landing/Login.js

```diff
@@ -3,69 +3,106 @@ import { Grid, withStyles, Typography, TextField } from "@material-ui/core";
 
 import styles from "Components/Form/LandingFormStyles";
 import StyledButton from "Components/Button/StyledButton";
+import GmailDialog from "./GmailDialog";
+import { checkForGmailToken } from "./gmailAuth";
 
 class Login extends Component {
   state = {
     email: "",
-    password: ""
+    password: "",
+    gmailDialogOpen: false,
   };
 
-  onChange = e => {
+  onChange = (e) => {
     this.setState({
-      [e.target.name]: e.target.value
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  onClick = async (e) => {
+    /*
+     * TO DO: Get user ID from login process / local storage
+     */
+    const userId = "5e84b3101bd834092a28464f";
+    const tokenExists = await checkForGmailToken(userId);
+
+    /*
+     *  Skip dialog
+     *  TO DO: Goto user's home page instead of returning true
+     */
+    if (tokenExists) return true;
+
+    // If the user hasn't authorised gmail access, launch dialog
+    // to prompt them to do it
+    this.setState({
+      gmailDialogOpen: true,
+    });
+  };
+
+  handleClose = () => {
+    this.setState({
+      gmailDialogOpen: false,
     });
   };
 
   render() {
     return (
-      <Grid
-        item
-        container
-        direction="column"
-        alignContent="center"
-        alignItems="center"
-        spacing={7}
-        className={this.props.classes.root}
-      >
-        <Grid item>
-          <Typography className={this.props.classes.header}>Login</Typography>
-        </Grid>
-
+      <span>
```

Comment:
It looks like you've added a `<span>` to wrap the Login form in. While this doesn't appear to be causing any issues, the `<Grid>` component should be the top-level component since it is an `item` that is part of a larger grid (the Landing page itself). You can just have the `<GmailDialog>` component inside the top-level `<Grid>`  component instead.

Reply:
Thanks. Fixed.